app/src/main/java/com/example/cleanarchitecture/activity/MainActivity.kt

```diff
@@ -1,12 +1,45 @@
 package com.example.cleanarchitecture.activity
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
 import com.example.cleanarchitecture.R
+import com.example.cleanarchitecture.databinding.ActivityMainBinding
+import com.example.cleanarchitecture.utils.Data
+import com.example.cleanarchitecture.utils.Event
+import com.example.cleanarchitecture.utils.Status
+import com.example.cleanarchitecture.viewmodel.CharacterViewModel
+import com.example.domain.entity.CharacterDataWrapperEntity
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private val viewModel by viewModel<CharacterViewModel>()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        viewModel.liveDataState.observe(this, { updateUI(it) })
+
+        binding.buttonMainActivityIdFetch.setOnClickListener { viewModel.onFetchCharactersClicked() }
+    }
+
+    private fun updateUI(characterData: Event<Data<CharacterDataWrapperEntity>>?) {
+        when (characterData?.peekContent()?.responseType) {
+            Status.SUCCESSFUL -> {
+                Toast.makeText(
```

Comment:
here you can do a fun how show this toast passing the string in the parameter

Reply:
done